core/unit_test/TestTimer.hpp

```diff
@@ -0,0 +1,52 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Timer.hpp>
+
+#include <thread>
+#include <type_traits>
+
+namespace {
+
+TEST(TEST_CATEGORY, timer) {
+  using namespace std::chrono_literals;
+
+  Kokkos::Timer t;
+  std::this_thread::sleep_for(5ms);
+  auto elapsed = t.seconds();
+  EXPECT_GE(elapsed, .005);
+  EXPECT_LE(elapsed, .01);
```

Comment:
What we usually do with such "fragile" tests is the following:
```suggestion
  Kokkos::Timer outer;
  Kokkos::Timer inner;
  std::this_thread::sleep_for(5ms);
  const auto elapsed_inner = inner.seconds();
  const auto elapsed_outer = outer.seconds();
  EXPECT_GE(elapsed_inner, .005);
  EXPECT_LT(elapsed_inner, elapsed_outer);
  EXPECT_LT(elapsed_outer, 1.);
```

See also https://github.com/uliegecsm/kokkos-utils/blob/d1f4033364858311f9001867452643286b078a8d/benchmarks/callbacks/benchmark_RegionTimer.cpp#L41-L43.

Reply:
if we wanted to be real anal we could use actual std::chromo instead of the outer timer. But I wouldn't mind just getting rid of the upper limit.

Reply:
I opted for just dropping the upped bound

Reply:
Not checking the upper bound means you might be returning garbage that is > 0.05 and > 0.015.

I guess that checking it returns less than 1 second costs nothing but makes it a little less brittle ?

Reply:
I mean, strict timing assertions can be very fragile due to system variability, but having at least a loose upper bound helps catch obvious failures without making the test flaky.

Reply:
I don't think that is worth our time.

Reply:
I think the timer is simple enough that the test we have now is sufficient.

Reply:
I guess since I didn't get a 2nd approval I have time to push some of the changes you wanted :/